components/dashboard_modules/SunnyTimeframe.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC } from 'react';
+
+import {
+  useFutureGraphData,
+  getNextThresholdIndex,
+  graphThreshold,
+  Value,
+} from 'lib/utils';
+
+const SunnyTimeframe: FC = () => {
+  const { data } = useFutureGraphData();
+  const nextThreshold = getNextThresholdIndex(
+    data?.forecast_values,
+    graphThreshold
+  );
```

Comment:
This computation isn't trivial, so while I wouldn't normally recommend it, maybe wrap this in a `useMemo`.
```suggestion
  const nextThreshold = getNextThresholdIndex(
    data?.forecast_values,
    graphThreshold
  );
```

Reply:
This component won't rerender much I don't think, so hold off on this for now.